docs/reference/rest-api/common-parms.asciidoc

```diff
@@ -209,22 +209,24 @@ tag::expand-wildcards[]
 `expand_wildcards`::
 +
 --
-(Optional, string) Controls what kind of indices that wildcard expressions can
-expand to. Multiple values are accepted when separated by a comma, as in
-`open,hidden`. Valid values are:
+(Optional, string)
+Type of index that wildcard expressions can match. If the request can target
+data streams, this argument determines whether wildcard expressions match
+hidden data streams. Supports comma-separated values, such as `open,hidden`.
+Valid values are:
 
 `all`::
-Expand to open and closed indices, including <<hidden-indices,hidden indices>>.
+Match any data stream or index, including <<hidden-indices,hidden>> ones.
 
 `open`::
-Expand only to open indices.
+Match non-hidden data streams and open, non-hidden indices.
```

Comment:
In my testing, `open` and `closed` match hidden indices and data streams though I'm not sure why.

Reply:
Thanks @danhermann. That sounds like a big regression, but I wasn't able to reproduce. But maybe my testing was wrong. Here's how I tested when running the latest master branch locally using `./gradlew run -Dtests.es.xpack.security.enabled=false`:

```
### 1. Create a non-hidden data stream.

PUT _index_template/visible-data-stream-template
{
  "index_patterns": [ "visible-data-stream" ],
  "data_stream": { },
  "priority": 200
}
PUT _data_stream/visible-data-stream

### 2. Create hidden data stream.

PUT _index_template/hidden-data-stream-template
{
  "index_patterns": [ "hidden-data-stream" ],
  "data_stream": {
    "hidden": true
  },
  "priority": 200
}
PUT _data_stream/hidden-data-stream

### 3. Get request with `expand_wildcards=open`.

GET *-data-stream?expand_wildcards=open

# Returns info for `visible-data-stream` only.

### 4. Get data streams with `expand_wildcards=open,hidden`.

GET *-data-stream?expand_wildcards=open,hidden

# Returns info for `visible-data-stream` and `hidden-data-stream`.

### 5. Create an open, hidden index.

PUT hidden-index
{
  "settings": {
    "index.hidden": true
  }
}

### 6. Create and close an index.

PUT closed-index
POST closed-index/_close

### 7. Get request with `expand_wildcards=closed`.

GET *-data-stream,*-index?expand_wildcards=closed

# Returns info for `closed-index` only.
```

Reply:
Hmm, interesting. I was testing primarily with `*` as the wildcard. With a more restrictive wildcard, it does appear to work as you've described.